crypto/bn/bn_lib.c

```diff
@@ -505,7 +505,8 @@ int bn2binpad(const BIGNUM *a, unsigned char *to, int tolen, endianess_t endiane
     /* Swipe through whole available data and don't give away padded zero. */
     atop = a->dmax * BN_BYTES;
     if (atop == 0) {
-        OPENSSL_cleanse(to, tolen);
+        if (to != NULL)
```

Comment:
is to==NULL an expected input to this function?

Reply:
Yes, it is possible. The function needs to return the length of the buffer.

Reply:
but why is it safe to assume to != NULL in the loop below?

Reply:
Hmm, good point! So the test failure was actually detecting a wrong call.

Reply:
So the testcase expects to return 0 from `BN_bn2binpad(BN_zero(), NULL, 0)` - Is that a correct or incorrect call? IMO keeping this if() here is harmless.

Reply:
FYI it's the test_bn2padded() in bntest.c.

Reply:
Hmm, changing that to `if (tolen != 0)` would keep it in sync with the loop below.

Reply:
Well, of course we could declare it a bug-fix that the case to==NULL did not return the required length,
without causing a fault?

Reply:
Sounds like a bug to me.

Reply:
I'd prefer the cleanse call to the conditional and `memset(3)`

Reply:
> I'd prefer the cleanse call to the conditional and `memset(3)`

I do not understand. The cleanse call is apparently breaking the fuzzer in #17237

Reply:
It looks more like the code the fuzzer was using to test things was broken....